### ISSUE TITLE
Repeat references are not resolved

### ISSUE THREAD

**Issue by fehguy** [swagger-api/swagger-parser#33]
Looking at this example:

https://gist.github.com/rbdixon/21fa977e4af15e7a0da7

repeating references are not properly resolved.

**Comment by fehguy** [comment]
available in 1.0.5-SNAPSHOT.

**Comment by alotau** [comment]
Is there an easy way to make use of this fix in either the codegen or (preferably) the editor projects?  I can't figure out how those projects reference this one.

**Comment by fehguy** [comment]
codegen references this version--if you're using the develop_2.0 branch of codegen, you should pick this up automatically.  Please give it a shot and report back if you have troubles

**Comment by alotau** [comment]
Tried against the master branch then the develop_2.0 branch of codegen. Same exception:

```java
java.lang.ClassCastException: com.wordnik.swagger.models.parameters.RefParameter cannot be cast to com.wordnik.swagger.models.parameters.BodyParameter
	at com.wordnik.swagger.codegen.DefaultCodegen.fromParameter(DefaultCodegen.java:939)
	at com.wordnik.swagger.codegen.DefaultCodegen.fromOperation(DefaultCodegen.java:792)
	at com.wordnik.swagger.codegen.DefaultGenerator.processOperation(DefaultGenerator.java:327)
	at com.wordnik.swagger.codegen.DefaultGenerator.processPaths(DefaultGenerator.java:299)
	at com.wordnik.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:156)
	at com.wordnik.swagger.codegen.cmd.Generate.run(Generate.java:77)
	at com.wordnik.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:33)
```
When switching between the branches and executing
```
mvn package
```
I can see the different versions of the parser being picked up.  For the develop_2.0 branch I see:
```
[INFO] swagger-parser-1.0.6-SNAPSHOT.jar already exists in destination.
```
during the tests.  So I think it is picking up the right jar (even though you note 1.0.5 above).

Am I missing a key step or is this fix not really available?

Thanks for the help.

**Comment by fehguy** [comment]
Hmmm, it should be.  I will reply later today after testing.

**Comment by alotau** [comment]
I think I got it to work.  There were a couple other warnings spit out with codegen that I addressed and then parser 1.0.5 seemed to be happy.  If it's useful to you it had to do with having a type: number with format: int32.  Changed type to integer to fix the warnings, then the parsing error went away.  Thanks for taking the time to look into this.

**Comment by fehguy** [comment]
looks like we have it working
